Add Alibaba Cloud `DashScopeProvider` and support audio input for Qwen Omni

Pavanmanikanta98 opened the pull request:
Fixes #3530

  Key Changes:

   * Updated Profile: Added openai_audio_input_encoding: Literal['base64', 'uri'] to OpenAIModelProfile.
       * 'base64' (default): Maintains strict OpenAI compliance.
       * 'uri': Enables Data URI formatting for providers like Qwen Omni.
   * Updated Model Logic: Modified OpenAIChatModel._map_user_prompt to respect this setting.
       * For BinaryContent: Uses item.data_uri when encoding is 'uri'.
       * For AudioUrl: Manually constructs the Data URI with the correct MIME type (e.g., audio/mpeg for mp3) when
         encoding is 'uri'.
   * New Tests: Added tests/models/test_openai_audio.py covering both default and URI encoding scenarios for both binary content and audio URLs.

DouweM commented on pydantic_ai_slim/pydantic_ai/providers/qwen.py:
```diff
@@ -0,0 +1,86 @@
+    @property
+    def base_url(self) -> str:
+        # Using the international endpoint by default as it's more standard for global users
+        # Users in China region can override this via passing `openai_client` or implementing logic to check region
```
Can we take a `base_url` argument? And mention this in the docs

DouweM commented on docs/models/openai.md:
```diff
@@ -309,6 +309,50 @@ agent = Agent(model)
 
+### DashScope
+
+To use Qwen models via [Alibaba Cloud DashScope](https://www.alibabacloud.com/en/product/modelstudio), you can set the `DASHSCOPE_API_KEY` environment variable and use [`DashScopeProvider`][pydantic_ai.providers.dashscope.DashScopeProvider] by name:
```
Is `DashScopeProvider` really the most appropriate/recognizable name? I see DashScope mentioned only once on https://www.alibabacloud.com/en/product/modelstudio, so maybe it should just be `AlibabaProvider`?

Pavanmanikanta98 commented:
@DouweM

  I've renamed DashScopeProvider to AlibabaProvider (using alibaba: prefix) and updated the documentation to reference "Alibaba Cloud Model Studio (DashScope)" as discussed. `DASHSCOPE_API_KEY` is preserved for consistency with the official docs. Ready for review
